From 1ff1546610cdb6df294d02b4bb122752d2342fd1 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 10 Jun 2022 13:46:26 +0300
Subject: [PATCH] [SW-0] common: fix init desc progress bits for erom

The init desc progress bits for erom doesn't match with spi
so fix it.

Change-Id: I69d0a20a6659e197498f2c64b85acf1c8a9294ce
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 81b35ec..170885c 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -292,7 +292,7 @@
 
 	if (!boot_desc_init(HL_FW_BOOT_PTR_EROM)) {
 		arcboot_rom_status_update(
-				PROGRESS_ARCBOOT_EROM_BOOT_DESC_ERROR);
+				PROGRESS_INIT_EROM_BOOT_DESC_FAIL);
 		return;
 	}
 
@@ -317,7 +317,7 @@
 	   EFUSE_SYSCFG_ENABLE_TPM_MASK) != 0) {
 		if (!boot_desc_init(HL_FW_BOOT_PTR_EROM)) {
 			arcboot_rom_status_update(
-				PROGRESS_ARCBOOT_EROM_BOOT_DESC_ERROR);
+				PROGRESS_INIT_EROM_BOOT_DESC_FAIL);
 			return;
 		}
 		ext_rom_load_erom_tpm();
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 4a7df1d..2d0584c 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -708,7 +708,7 @@
 	if (!security_efuse_tpm_loc()) {
 		if (!boot_desc_init(HL_FW_BOOT_PTR_EROM)) {
 			arcboot_rom_status_update(
-					PROGRESS_ARCBOOT_EROM_BOOT_DESC_ERROR);
+					PROGRESS_INIT_EROM_BOOT_DESC_FAIL);
 			return;
 		}
 
diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 08215bf..230a34b 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -186,7 +186,7 @@
 
 	count = boot_desc_init(HL_FW_BOOT_PTR_EROM);
 	if (!count) {
-		arcboot_rom_status_update(PROGRESS_ARCBOOT_EROM_BOOT_DESC_ERROR);
+		arcboot_rom_status_update(PROGRESS_INIT_EROM_BOOT_DESC_FAIL);
 		goto rom_recovery;
 	}
 
diff --git a/common/boot_desc.c b/common/boot_desc.c
index 1ab9195..092819d 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -77,7 +77,7 @@
 		return p_boot_ptr->count;
 
 	if (img == HL_FW_BOOT_PTR_EROM) {
-		dma_sts_base = PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_START;
+		dma_sts_base = PROGRESS_INIT_EROM_BOOT_DESC_DMA_START;
 	} else {
 		if (arcboot_rom_global.validation_pass)
 			goto skip_boot_desc_process;
diff --git a/include/arcboot.h b/include/arcboot.h
index 8e87eee..4eac596 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -220,60 +220,61 @@
 	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 78,
 	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 79,
 	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 80,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_START = 81,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_UNALIGNED_DEST = 82,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 83,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_ERROR = 84,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_FINISH = 85,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 86,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_VERIFICATION_FAIL = 87,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_ERROR = 88,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 89,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 90,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 91,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 92,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 93,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 94,
-	PROGRESS_ARCBOOT_EROM_PCIE = 95,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 96,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_CRC_FAILED = 97,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_DEST = 98,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_IMG_SIZE = 99,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_ERROR = 100,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 101,
-	PROGRESS_ARCBOOT_EROM_PCIE_VERIFICATION_FAIL = 102,
-	PROGRESS_ARCBOOT_EROM_PCIE_START = 103,
-	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 104,
-	PROGRESS_ARCBOOT_EROM_RECOVERY = 105,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 106,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_CRC_FAILED = 107,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_DEST = 108,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_IMG_SIZE = 109,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_ERROR = 110,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 111,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_VERIFICATION_FAIL = 112,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 113,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 114,
-	PROGRESS_ARCBOOT_EROM_TPM = 115,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 116,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_CRC_FAILED = 117,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_DEST = 118,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_IMG_SIZE = 119,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_ERROR = 120,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 121,
-	PROGRESS_ARCBOOT_EROM_TPM_VERIFICATION_FAIL = 122,
-	PROGRESS_ARCBOOT_EROM_TPM_START = 123,
-	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 124,
-	PROGRESS_ARCBOOT_EROM_IMG0 = 125,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 126,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 127,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 128,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 129,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 130,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 131,
-	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 132,
-	PROGRESS_ARCBOOT_EROM_IMG0_START = 133,
-	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 134,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 81,
+	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 82,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 83,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 84,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 85,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 86,
+	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 87,
+	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 88,
+	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 89,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 90,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 91,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 92,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 93,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 94,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 95,
+	PROGRESS_ARCBOOT_EROM_PCIE = 96,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 97,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_CRC_FAILED = 98,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_DEST = 99,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_IMG_SIZE = 100,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_ERROR = 101,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 102,
+	PROGRESS_ARCBOOT_EROM_PCIE_VERIFICATION_FAIL = 103,
+	PROGRESS_ARCBOOT_EROM_PCIE_START = 104,
+	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 105,
+	PROGRESS_ARCBOOT_EROM_RECOVERY = 106,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 107,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_CRC_FAILED = 108,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_DEST = 109,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_IMG_SIZE = 110,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_ERROR = 111,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 112,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_VERIFICATION_FAIL = 113,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 114,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 115,
+	PROGRESS_ARCBOOT_EROM_TPM = 116,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 117,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_CRC_FAILED = 118,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_DEST = 119,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_IMG_SIZE = 120,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_ERROR = 121,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 122,
+	PROGRESS_ARCBOOT_EROM_TPM_VERIFICATION_FAIL = 123,
+	PROGRESS_ARCBOOT_EROM_TPM_START = 124,
+	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 125,
+	PROGRESS_ARCBOOT_EROM_IMG0 = 126,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 127,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 128,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 129,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 130,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 131,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 132,
+	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 133,
+	PROGRESS_ARCBOOT_EROM_IMG0_START = 134,
+	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 135,
 };
 
 enum security_location {
